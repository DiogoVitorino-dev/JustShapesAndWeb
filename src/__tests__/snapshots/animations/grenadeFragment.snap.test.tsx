import { act, render, waitFor } from "@testing-library/react-native";

import GrenadeFragment from "@/animations/attacks/grenade/grenadeFragment";

describe("Grenade Fragment - Snapshot animation test", () => {
  beforeEach(() => jest.useFakeTimers());
  afterEach(() => jest.useRealTimers());
  it("should render correctly", async () => {
    const tree = render(
      <GrenadeFragment angleDirection={90} start={false} />,
    ).toJSON();
    await waitFor(() =>
      act(() => {
        jest.advanceTimersByTime(5000);
      }),
    );
    expect(tree).toMatchSnapshot();
  });
});
