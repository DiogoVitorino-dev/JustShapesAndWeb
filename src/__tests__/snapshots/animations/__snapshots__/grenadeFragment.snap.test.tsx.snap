// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grenade Fragment - Snapshot animation test should render correctly 1`] = `
<View
  collapsable={false}
  jestAnimatedStyle={
    {
      "value": {
        "borderRadius": 5,
        "height": 10,
        "left": 0,
        "opacity": 0,
        "top": 0,
        "width": 10,
      },
    }
  }
  style={
    {
      "backgroundColor": "rgb(255, 64, 20)",
      "borderRadius": 5,
      "height": 10,
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": 10,
    }
  }
  testID="circleModel"
/>
`;
